metricbeat/autodiscover/builder/hints/metrics_test.go

```diff
@@ -548,13 +548,46 @@ func TestGenerateHints(t *testing.T) {
 				},
 			},
 		},
+		{
+			message: "exclude/exclude in metrics filters are parsed as a list",
+			event: bus.Event{
+				"host": "1.2.3.4",
+				"hints": common.MapStr{
+					"metrics": common.MapStr{
+						"module":    "prometheus",
+						"namespace": "test",
+						"hosts":     "${data.host}:9090",
+						"metrics_filters": common.MapStr{
+							"exclude": "foo, bar",
+							"include": "xxx, yyy",
+						},
+					},
+				},
+			},
+			len: 1,
+			result: []common.MapStr{
+				{
+					"module":     "prometheus",
+					"namespace":  "test",
+					"metricsets": []string{"collector"},
+					"timeout":    "3s",
+					"period":     "1m",
+					"enabled":    true,
+					"hosts":      []interface{}{"1.2.3.4:9090"},
+					"metrics_filters": map[string]interface{}{
+						"exclude": []interface{}{"foo", "bar"},
+						"include": []interface{}{"xxx", "yyy"},
+					},
+				},
+			},
+		},
 	}
 	for _, test := range tests {
 		mockRegister := mb.NewRegister()
-		mockRegister.MustAddMetricSet("mockmodule", "one", NewMockMetricSet, mb.DefaultMetricSet())
```

Comment:
Any reason for removing this?

Reply:
oops mistakenly removed this line, added back.

Reply:
👍🏼